x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/pipeline_processors_editor.test.tsx

```diff
@@ -252,5 +253,66 @@ describe('Pipeline Editor', () => {
       expect(data.processors).toEqual([testProcessors.processors[1], testProcessors.processors[2]]);
       expect(data.on_failure).toEqual([testProcessors.processors[0]]);
     });
+
+    it('shows user provided descriptions rather than default descriptions and default descriptions rather than no description', async () => {
+      const { actions, find } = testBed;
+
+      await actions.addProcessor('processors', 'test', { if: '1 == 1' });
+
+      const processorDescriptions = {
+        script: 'my script',
+        set: 'Sets value of "test" to "test"',
+        test: 'No description',
+      };
+
+      const createAssertForProcessor = (processorIndex: string) => ({
+        description,
+        descriptionVisible,
+      }: {
+        description: string;
+        descriptionVisible: boolean;
+      }) => {
+        expect(find(`processors>${processorIndex}.inlineTextInputNonEditableText`).text()).toBe(
+          description
+        );
+        expect(
+          (find(`processors>${processorIndex}.pipelineProcessorItemDescriptionContainer`).props()
+            .className as string).includes('--displayNone')
+        ).toBe(!descriptionVisible);
+      };
+
+      const assertScriptProcessor = createAssertForProcessor('0');
+      const assertSetProcessor = createAssertForProcessor('2');
```

Comment:
Sure, I skipped index 1 because I wanted to get a sample of 3 different cases:

1. Processor with a custom description
2. Processor with a default description
3. Processor with no description

So I selected 3 processors from the list and they happen to be at those positions in the array of processors. Perhaps these variables can be renamed to better reflect that 👍🏻